File: convalida-validators/src/main/java/convalida/validators/ConfirmEmailValidator.java
package convalida.validators;

import android.widget.EditText;

/**
 * @author Wellington Costa on 07/02/18.
 */
public class ConfirmEmailValidator extends AbstractValidator {

    private EditText emailEditText;

    public ConfirmEmailValidator(
            EditText emailEditText,
            EditText confirmEmailEditText,
            String errorMessage,
            boolean autoDismiss) {
        super(confirmEmailEditText, errorMessage, autoDismiss);
        this.emailEditText = emailEditText;
    }

    @Override public boolean isNotValid(String value) {
        return !emailEditText.getText().toString().equals(value);
    }

}

File: convalida-validators/src/main/java/convalida/validators/ConfirmEmailValidator.kt
package convalida.validators

import android.widget.EditText

/**
 * @author Wellington Costa on 07/02/18.
 */
class ConfirmEmailValidator(val emailField: EditText, confirmEmailField: EditText, errorMessage: String) : AbstractValidator(confirmEmailField, errorMessage) {

    override fun isNotValid(value: String): Boolean {
        return emailField.text.toString() != value
    }

}
